x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -556,6 +559,21 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
         return Collections.singletonMap(InferenceProcessor.TYPE, inferenceFactory);
     }
 
+    @Override
+    public void loadExtensions(ExtensionLoader loader) {
+        loader.loadExtensions(MachineLearningExtension.class).forEach(machineLearningExtensions::add);
```

Comment:
Assuming `machineLearningExtensions` is renamed `machineLearningExtension` and turned into a `SetOnce`:

```suggestion
        loader.loadExtensions(MachineLearningExtension.class).forEach(machineLearningExtension::set);
        if (machineLearningExtension.get() == null) {
            machineLearningExtension.set(new DefaultMachineLearningExtension());
        }
```